lib/spack/spack/build_environment.py

```diff
@@ -475,6 +476,33 @@ def static_to_shared_library(static_lib, shared_lib=None, **kwargs):
 
     m.static_to_shared_library = static_to_shared_library
 
+    def echoize(fn):
+        @functools.wraps(fn)
+        def wrapper(*args, **kwargs):
+            if hasattr(pkg, 'logger'):
```

Comment:
Had to add this because `tty` gets called frequently inside `lib/spack/spack/package.py`, before `self.logger` has been created.